src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -3712,6 +3712,43 @@ private static void AssertContents(string expectedValue, ArrayBufferWriter<byte>
 #endif
                     ));
         }
+
+        [Fact]
+        // Issue #39930
+        public static void Disposing_On_Multiple_Threads_Should_Not_Return_To_ArrayPool_Twice()
+        {
+            Action<object> disposeAction = (object document) => ((JsonDocument)document).Dispose();
+
+            // Create a bunch of parallel tasks that call Dispose several times on the same object.
+            Task[] tasks = new Task[100];
+            int count = 0;
+            for (int j = 0; j < 10; j++)
+            {
+                JsonDocument document = JsonDocument.Parse("123" + j);
+                for (int i = 0; i < 10; i++)
+                {
+                    tasks[count] = new Task(disposeAction, document);
+                    tasks[count].Start();
+
+                    count++;
+                }
+            }
+
+            Task.WaitAll(tasks);
+
+            // When ArrayPool gets corrupted, the Rent method might pull an already rented array, which is incorrect.
```

Comment:
nit: What do you mean by pull here?

```suggestion
            // When ArrayPool gets corrupted, the Rent method might return an already rented array, which is incorrect.
```

Reply:
you are right, return would be more appropriate, thanks for the suggestion.